includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -445,7 +445,7 @@ public function process_payment( $order_id ) {
 		$payment_needed            = 0 < $converted_amount;
 		$selected_upe_payment_type = $this->stripe_id;
 		$payment_type              = $this->is_payment_recurring( $order_id ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
-		$save_payment_method       = $payment_type->equals( Payment_Type::RECURRING() ) || ! empty( $_POST[ 'wc-' . self::GATEWAY_ID . '_' . $this->stripe_id . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$save_payment_method       = $payment_type->equals( Payment_Type::RECURRING() ) || ! empty( $_POST[ 'wc-' . self::GATEWAY_ID . '_' . $this->stripe_id . '-new-payment-method' ] ) || ! empty( $_POST[ 'wc-' . $this->stripe_id . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
This solution worked for me...but I believe this `POST` key is different in the blocks checkout because when we register the blocks payment method [the name parameter](https://github.com/Automattic/woocommerce-payments/blob/fix/5260-blocks-checkout-save-upe-methods/client/checkout/blocks/upe.js#L67) is set to the Stripe ID, rather than the gateway ID. You can see that previously we used to use the gateway ID--prefixed with `woocommerce_payments`--instead in the regular pre-partition UPE. 

I think if we used `upeMethods[ upeName ]` instead of just `upeName` for this name field, that would instead use the same gateway ID for both blocks and shortcode checkouts. This also would have the benefit of differentiating this gateway from other non-WCPay gateways that may also register `sepa_debit` as a name field, for whatever reason. 

Let me know your thoughts, @mdmoore--especially if this might have already been an idea that you contemplated and rejected. Cheers!